freqtrade/rpc/telegram.py

```diff
@@ -242,6 +242,35 @@ def _format_buy_msg(self, msg: Dict[str, Any]) -> str:
         message += ")`"
         return message
 
+    def _format_buy_fill_msg(self, msg: Dict[str, Any]) -> str:
+        if self._rpc._fiat_converter:
+            msg['stake_amount_fiat'] = self._rpc._fiat_converter.convert_amount(
+                msg['stake_amount'], msg['stake_currency'], msg['fiat_currency'])
+        else:
+            msg['stake_amount_fiat'] = 0
+
+        msg['currency'] = msg['pair'].split('/')[0]
```

Comment:
err - no.
Getting stake-currency like this is something we moved away from quite some time (it's fluky, and only works with certain base assumptions on the exchange used) - and will no longer work when futures/shorts land - so this is introducing a bug in my view.

If you really need to use this, then please add it to the originating dictionaries (in `freqtradebot.py`) - calculated with `exchange.get_pair_quote_currency(pair)` and/or `exchange.get_pair_base_currency(pair)`.

I'd also not call that key "currency" - but "asset" (or `base_currency` - or something similar more generic) - as some "coins" are not really meant to be currencies, but are tokens (Like the fan-tokens we have), and "currency" is quite generic, so i'd not be sure if you now mean fiat-currency, stake-currency, or the coin we're buying

Reply:
Not a necessity, will remove for now